Update changelog (#357)
NitorCreations/nflow: CHANGELOG.md

- Expose wakeup via REST-API
- Update database indices to match workflow instance polling code
- Add new configuration option `nflow.db.disable_batch_updates`

> - Workarounds for Galera oddities to make nflow more stable

Reviewer: ```suggestion
- Workarounds for Galera oddities to make nFlow more stable when Galera is used
```

Reviewer 2: I don't think we did anything except that above `nflow.db.disable_batch_updates` actually. If not documenting in Wiki counts?

  - Update database indices to match current workflow instance polling code.
  - Create indices for foreign keys in MS SQL database.
  - Fix create database scripts to work with empty database.

>   - Add `nflow.db.disable_batch_updates` (default `false`) configuration parameter to make it possible to force use of multiple updates even if batch updates are supported by the database. This is a needed forkaround on some Galera systems that cannot handle batches correctly.

Reviewer: ```suggestion
  - Add `nflow.db.disable_batch_updates` (default `false`) configuration parameter to make it possible to force use of multiple updates even if batch updates are supported by the database. This is needed on some Galera systems that cannot handle batch updates correctly.
```

- Improve error logging in WorkflowStateProcessor.
- Replace FindBugs with SpotBugs.
- Drop index from main table that was only used for archiving purposes.

> - Increase workflow and action identifiers in code to 64 bits for future proofing. Database schema is not changed for now, but the columns can be altered later (to bigints) if needed.

Reviewer: ```suggestion
- Increase workflow instance and action identifiers in code to 64 bits for future proofing. Database schema is not changed for now, but the columns can be altered later (to bigints) if needed.
```

> - Increase workflow and action identifiers in code to 64 bits for future proofing. The SQL table definitions stay the same for now, but no code changes are needed if database is altered to use bigint identifiers in large systems. The default 31 bits should be enough for 10 workflows per second with each having 20 steps for 8 years constant load.

Reviewer: ```suggestion
- Increase workflow and action identifiers in code to 64 bits for future proofing. Database schema is not changed for now, but the columns can be altered later (to bigints) if needed.
```

- Improve error logging in WorkflowStateProcessor.
- Replace FindBugs with SpotBugs.
- Drop index from main table that was only used for archiving purposes.

> - Increase workflow and action identifiers in code to 64 bits for future proofing. The SQL table definitions stay the same for now, but no code changes are needed if database is altered to use bigint identifiers in large systems. The default 31 bits should be enough for 10 workflows per second with each having 20 steps for 8 years constant load.

Reviewer 2: Here the caveat is that multi-master databases jump over a lot of indices. In our galera the gap is 3, so 1, 4, 7, 10...
